sdk/core/azure-core/src/http/http.cpp

```diff
@@ -156,7 +156,7 @@ void Azure::Core::_detail::InsertHeaderWithValidation(
   {
```

Comment:
This is a pretty substantial perf hit for every header for every operation invocation.
I didn't know we were doing this; I don't think I would have had it.

Reply:
I agree, but we need to let the caller know they have some bug in their code.

I wish there was a mechanism for us to mark APIs like `response.SetHeader()` internal-only. Right now this is a public API, that users can call and pass in any string. Validating that is useful. It's possible the users pass in some untrusted strings here.

Alternatively, we should consider trying to scope down the API surface and limit the several ways folks pass in data.

Let's discuss it separate from this PR.

Reply:
I never liked this T.T , and still don't like it.

Headers most likely added by service clients. The service client should catch an invalid header during testing the sdk and update the header name to be valid.

Same thing could apply with end-users, and expect them to use the service response as enough to realize they set an invalid header.

An application running on production won't be able to do anything about this exception.